src/ui/preview/Preview.vue

```diff
@@ -108,24 +130,58 @@ export default {
                 return;
             }
 
+            // if there is an existing view element, then it's currently being viewed
+            // we'll add it back to the parent in the meantime
+            if (this.existingViewElement) {
+                this.addExistingViewBackToParent();
+            }
+
             this.clear();
             this.viewKey = view.key;
             this.currentView = view;
-            this.viewContainer = document.createElement('div');
-            this.viewContainer.classList.add('l-angular-ov-wrapper');
-            this.$refs.objectView.append(this.viewContainer);
+            this.initializeViewContainer();
             this.view = this.currentView.view(this.domainObject, this.objectPath);
-
-            this.getActionsCollection();
+            this.getActionsCollection(this.view);
             this.view.show(this.viewContainer, false, this.viewOptions);
             this.initObjectStyles();
         },
-        getActionsCollection() {
+        showExistingView() {
```

Comment:
I _suspect_ that we could simplify this Vue component a little bit by just combining this method with `setView`. See specific comments.

Reply:
agreed!